src/sync-pending-events.ts

```diff
@@ -152,7 +152,10 @@ export class EventTracker {
           await db.updateLastMonitorBlock(r, e);
         }
       }
-      await db.updatelastCheckedBlockNumber(pastEvents.breakpoint);
+      if (pastEvents.breakpoint > lastCheckedBlockNumber){
```

Comment:
let breakpoint = await getValidBlockNumber(this.source, lastCheckedBlockNumber, pastEvents.breakpoint);
if (breakpoint)
    await db.updatelastCheckedBlockNumber(breakpoint);